[Bug] API call for users/extensions panics

### What is the problem?

When running `twitch api get users/extensions -q user_id=<user id>`, the cli panics.

### Operating System

All

### Architecture Version (x86, x64, arm, etc)

All

### Steps to reproduce

1. Run `twitch api get users/extensions -q user_id=57047445`
Any User ID should do.

### Relevant log output

```shell
> twitch api get users/extensions -q user_id=57047445
panic: interface conversion: interface {} is map[string]interface {}, not []interface {}

goroutine 1 [running]:
github.com/twitchdev/twitch-cli/internal/api.NewRequest({0xaa8a42, 0x3}, {0xc000127d48, 0x11}, {0xc0000900e0, 0x1, 0xc0000a2500?}, {0xf698a8, 0x0, 0x0}, ...)
        /go/src/github.com/twitchdev/twitch-cli/internal/api/api.go:178 +0x11c5
github.com/twitchdev/twitch-cli/cmd.cmdRun(0x98c8a0?, {0xc0000a61e0?, 0x3?, 0x3?})
        /go/src/github.com/twitchdev/twitch-cli/cmd/api.go:127 +0x2b0
github.com/spf13/cobra.(*Command).execute(0x98c8a0, {0xc0000a61b0, 0x3, 0x3})
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856 +0x663
github.com/spf13/cobra.(*Command).ExecuteC(0x98da20)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960 +0x39d
github.com/spf13/cobra.(*Command).Execute(...)
        /go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/twitchdev/twitch-cli/cmd.Execute()
        /go/src/github.com/twitchdev/twitch-cli/cmd/root.go:26 +0x25
main.main()
        /go/src/github.com/twitchdev/twitch-cli/main.go:16 +0x50
```

--- a/internal/api/api.go
+++ b/internal/api/api.go
@@ -111,13 +111,19 @@ func NewRequest(method string, path string, queryParameters []string, body []byt
 				fmt.Printf("Error unmarshalling body: %v", err)
 				return
 			}
+
+			var cursor string = ""
+			if extensionsBody.Pagination != nil {
+				cursor = *extensionsBody.Pagination
+			}
+
 			apiResponse = models.APIResponse{
 				Data:    extensionsBody.Data,
 				Status:  extensionsBody.Status,
 				Error:   extensionsBody.Error,
 				Message: extensionsBody.Message,
 				Pagination: &models.APIPagination{
-					Cursor: *extensionsBody.Pagination,
+					Cursor: cursor,
 				},
 			}
 		} else {

--- a/internal/api/api.go
+++ b/internal/api/api.go
@@ -35,7 +35,7 @@ func NewRequest(method string, path string, queryParameters []string, body []byt
 	var err error
 	var cursor string
 
-	isExtensionsEndpoint := false // https://github.com/twitchdev/twitch-cli/issues/157
+	isExtensionsLiveEndpoint := false // https://github.com/twitchdev/twitch-cli/issues/157
 
 	data.Data = make([]interface{}, 0)
 	client, err := GetClientInformation()
@@ -104,7 +104,7 @@ func NewRequest(method string, path string, queryParameters []string, body []byt
 		}
 		if strings.Contains(u.String(), "extensions/live") {
 			// https://github.com/twitchdev/twitch-cli/issues/157
-			isExtensionsEndpoint = true
+			isExtensionsLiveEndpoint = true
 			var extensionsBody models.ExtensionAPIResponse
 			err = json.Unmarshal(resp.Body, &extensionsBody)
 			if err != nil {
@@ -181,12 +181,17 @@ func NewRequest(method string, path string, queryParameters []string, body []byt
 		data.Data = make([]interface{}, 0)
 	}
 	// handle json marshalling better; returns empty slice vs. null
-	if !strings.Contains(path, "schedule") && len(data.Data.([]interface{})) == 0 && data.Error == "" {
+	_, isInterface := data.Data.([]interface{})
+	if isInterface && !strings.Contains(path, "schedule") && len(data.Data.([]interface{})) == 0 && data.Error == "" {
 		data.Data = make([]interface{}, 0)
+	}
+	_, isStringMap := data.Data.(map[string]any)
+	if isStringMap && !strings.Contains(path, "schedule") && len(data.Data.(map[string]any)) == 0 && data.Error == "" {
+		data.Data = make(map[string]any, 0)
 	}
 
 	var d []byte
-	if isExtensionsEndpoint {
+	if isExtensionsLiveEndpoint {
 		extensionBody := models.ExtensionAPIResponse{
 			Data:       data.Data,
 			Pagination: &data.Pagination.Cursor,

--- a/internal/models/api.go
+++ b/internal/models/api.go
@@ -3,7 +3,7 @@
 package models
 
 type APIResponse struct {
-	Data       interface{}               `json:"data,omitempty"`
+	Data       any                       `json:"data,omitempty"`
 	Pagination *APIPagination            `json:"pagination,omitempty"`
 	Error      string                    `json:"error,omitempty"`
 	Status     int                       `json:"status,omitempty"`
@@ -22,7 +22,7 @@ type BitsLeaderboardDateRange struct {
 	EndedAt   string `json:"ended_at"`
 }
 
-type ExtensionAPIResponse struct {
+type ExtensionAPIResponse struct { // extensions/live
 	Data       interface{} `json:"data,omitempty"`
 	Pagination *string     `json:"pagination,omitempty"`
 	Error      string      `json:"error,omitempty"`

